SU2_CFD/src/interfaces/cht/CConjugateHeatInterface.cpp

```diff
@@ -104,95 +93,112 @@ void CConjugateHeatInterface::GetDonor_Variable(CSolver *donor_solution, CGeomet
   dist = sqrt(dist2);
   Area = sqrt(Area);
 
-  /*--- Retrieve temperature solution (later set is as the first donor variable) and its gradient ---*/
-
-  dTdn = 0.0;
+  /*--- Retrieve temperature solution and its gradient ---*/
 
   if (compressible_flow) {
 
-    Twall   = donor_solution->node[Point_Donor]->GetPrimitive(0)*Temperature_Ref;
-    Tnormal = donor_solution->node[PointNormal]->GetPrimitive(0)*Temperature_Ref;
+    Twall   = donor_solution->node[Point_Donor]->GetPrimitive(0);
+    Tnormal = donor_solution->node[PointNormal]->GetPrimitive(0);
 
     dTdn = (Twall - Tnormal)/dist;
   }
   else if (incompressible_flow) {
 
-    Twall   = donor_solution->node[Point_Donor]->GetTemperature()*Temperature_Ref;
-    Tnormal = donor_solution->node[PointNormal]->GetTemperature()*Temperature_Ref;
+    Twall   = donor_solution->node[Point_Donor]->GetTemperature();
+    Tnormal = donor_solution->node[PointNormal]->GetTemperature();
 
     dTdn = (Twall - Tnormal)/dist;
   }
-  else if (flow || heat_equation) {
-    Twall   = donor_solution->node[Point_Donor]->GetSolution(0)*Temperature_Ref;
-    Tnormal = donor_solution->node[PointNormal]->GetSolution(0)*Temperature_Ref;
+  else if (heat_equation) {
+
+    Twall   = donor_solution->node[Point_Donor]->GetSolution(0);
+    Tnormal = donor_solution->node[PointNormal]->GetSolution(0);
 
+    // TODO: Check if these improve accuracy, if needed at all
     //    for (iDim = 0; iDim < nDim; iDim++) {
     //      dTdn += (Twall - Tnormal)/dist * (Edge_Vector[iDim]/dist) * (Normal[iDim]/Area);
     //    }
 
     dTdn = (Twall - Tnormal)/dist;
   }
   else {
-    cout << "WARNING: Transfer of conjugate heat variables is called with non-supported donor solver!" << endl;
+
+    SU2_MPI::Error("Transfer of conjugate heat variables failed (non-supported donor solver).", CURRENT_FUNCTION);
   }
 
-  /*--- Calculate the heat flux density (temperature gradient times thermal conductivity)
-   *--- and set it as second donor variable ---*/
+  /*--- Calculate the heat flux density (temperature gradient times thermal conductivity) ---*/
+
   if (compressible_flow) {
 
-    iPoint = donor_geometry->vertex[Marker_Donor][Vertex_Donor]->GetNode();
+    su2double Gamma         = donor_config->GetGamma();
+    su2double Gas_Constant  = donor_config->GetGas_ConstantND();
+    su2double Cp            = (Gamma / (Gamma - 1.0)) * Gas_Constant;
+
+    Prandtl_Lam             = donor_config->GetPrandtl_Lam();
+    laminar_viscosity       = donor_solution->node[Point_Donor]->GetLaminarViscosity(); // TDE check for consistency
+    Cp                      = (Gamma / (Gamma - 1.0)) * Gas_Constant;
```

Comment:
@economon This is now completely analogue to the code in `CNSSolver::Friction_Forces`. Do you agree with it?

Reply:
Perhaps it is best to have the flow solvers compute the information that you need here just once and store it (maybe we can use what is computed in Friction_Forces() or something similar), and only have the interface class access it from the corresponding solver (compressible vs incompressible). It could be dangerous to have two implementations of the same thing - if one changes and the other doesn't, the functionality breaks. It would be a lot easier to maintain, and simpler, to define it just once. Also, these quantities can be different between compressible and incompressible.

Reply:
Yes it's just about the compressible case. 
For the incompressible one, I get a non-dimensionalized thermal conductivity from `donor_solution->node[iPoint]->GetThermalConductivity()` to then compute everything and finally use `donor_config->GetHeat_Flux_Ref()` to dimensionalize.
I am just not so much familiar with the compressible solver, but if it's sure that I could e.g. get the thermal conductivity always from `FluidModel` I could replace the computation (which is unsafe, as you pointed out) there by such a solver call.